cartographer_rviz/cartographer_rviz/drawable_submap.cc

```diff
@@ -73,7 +80,12 @@ DrawableSubmap::DrawableSubmap(const int trajectory_id, const int submap_index,
   material_->setCullingMode(Ogre::CULL_NONE);
   material_->setDepthBias(-1.f, 0.f);
   material_->setDepthWriteEnabled(false);
-  scene_node_->attachObject(manual_object_);
+  visibility_ = ::cartographer::common::make_unique<::rviz::BoolProperty>(
```

Comment:
If I remember correctly, I had a good reason to introduce the unique_ptr - because [here](https://github.com/googlecartographer/cartographer_ros/pull/294/commits/cd50b20a5792bdb684c769571dcc54d257a4fe4e#diff-549491b98125fc0df2c02d4d2023a371R126) I purge old submaps in case I detect there's now less of them, which happens when you relaunch the Cartographer node. The Boolproperty objects have the Category object as their Qt parent, and not the DrawableSubmap - so they will survive the destruction, because Qt will not destroy them when we destroy the DrawableSubmap. So it was either adding a delete in the DrawableSubmap destructor, or using a unique pointer, and I went with the latter.